include/git2/filter.h

```diff
@@ -159,6 +160,18 @@ GIT_EXTERN(int) git_filter_list_apply_to_file(
 	git_repository *repo,
 	const char *path);
 
+/**
+ * Apply a filter list to the contents of a git_file on disk
+ *
+ * @param out buffer into which to store the filtered file
+ * @param filters the list of filters to apply
+ * @param fd the file to filter
+ */
+GIT_EXTERN(int) git_filter_list_apply_to_git_file(
+	git_buf *out,
+	git_filter_list *filters,
+	git_file fd);
```

Comment:
I don't want to expose an API where people give us file descriptors.  `git_file` should be an internal abstraction.

I think that adding this helper method is reasonable, but it looks like we can safely keep it an internal function, does that seem correct?

Reply:
This makes sense, I've moved the 2 functions to the private API